### ISSUE TITLE
Suggestions for improving related resources tab

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#1444]
~~1. When a resource is added to a collection make the change take effect immediately without requiring the save changes button.  This would make the behavior similar to keywords.
2.  For removing a resource from a collection have an Are you sure check, similar to deleting a metadata element rather than just removing the resource and then waiting for save changes.~~
3.  When displaying relationships in the related resources tab list relationships consistently.  For example in the resource below, there is a derived from, isDataFor and Belongs to elements.  It seems that all of these should list the same information.  It is inconsistent to just give the GUID for some, but title, Owners etc for others.

![untitled](https://cloud.githubusercontent.com/assets/3456833/18414882/3f8404ae-7798-11e6-8638-64b2306c8b00.png)

4.  When adding relationships offer the user a resource picker control rather than a blank box where a URL has to be entered.  At least this should be possible for resources in HydroShare.  To enable relationships to resources outside of HydroShare a box to enter a URL may be needed.

**Comment by zhiyuli** [comment]
@dtarb @horsburgh 
Regarding (1) and (2), currently the workflow of adding and removing resources in collections are  consistent: user makes changes to the "Collection Contents" list, and then clicks the "Save" btn. One click on the 'Save' btn results in one hit to the backend, writing changes to django db and bag. Current implementation can significantly reduce backend load as user can submit several changes (like adding 5 new and removing 2 old resources ) in one time instead of hitting backend multiple times.

Regarding (3), are all the terms in Source and Relation designed for referencing HydroShare resources? If yes, then we could change them to list the same info as "Belongs to Collections" section; if no, that means user can save arbitrary values, url or non-url. Although it can be a url, it may not be a hydroshare resource url, in which case no owner or sharing status can be displayed.

**Comment by dtarb** [comment]
@zhiyuli on (1) and (2) are you designing to reduce back end load or to make it easy and simple for a user.  Easy and simple for a user must come first.  On (3) source and relation may be internal or external.  We should support both.  We should allow picking from a list for internal.  For external we may have to accept that owner and sharing status are blank.  We may have to enable a user to enter information (such as title) for external related resources to make them more presentable.

**Comment by zhiyuli** [comment]
@dtarb 
Sure. I would have (1) and (2) implemented in next PR for collection, dont want to hold current PR too long.

(3) is a general improvement to "Related Resources" tab. What is the priority? Also I may need help from @mauriel133

**Comment by zhiyuli** [comment]
pasting @horsburgh 's suggestions on (3) here:

I agree with @dtarb. If you want to open a new issue for this we could have Mauriel help. Basically, what is needed is a sentence/blurb to act as the section header for each relationship type. For example:

"isPartOf" --> "This resources belongs to the following collections:"
"isHostedBy" --> "This resource is hosted by the following system or organization:"
"isCopiedFrom" --> "This resource was copied from the following resource or source:"
"isExecutedBy" --> "This resource can be executed by the following resources:"
"isDataFor" --> "This resource is the data for the following publication:"

**Comment by mauriel133** [comment]
@zhiyuli, is there a branch I could implement the suggestion above to, or should I create a new one?

**Comment by zhiyuli** [comment]
@mauriel133 
There is no branch for this issue yet. Please create a new one if you are about to work on this.
@dtarb 
Regarding (3), I am still thinking more discussion are needed before starting implementation. Especially for "referencing internal resources", what list should we provide for users to select from? any requirement on permission? This is not just front end improvement as "referencing internal resources' in Relation and Source is kind of "collection resource type" behavior which needs changes to metadata class.

**Comment by zhiyuli** [comment]
@dtarb 
(1) and (2) have been implemented and pushed to riverdev, which are changes to collection. If they look good, we could modify this issue to focus on (3)  -- general improvements to "Related Resources" tab.

**Comment by dtarb** [comment]
OK.  Implementation of (1) and (2) looks good, so I changed the title and added strikethrough to indicate that (1) and (2).

**Comment by dtarb** [comment]
@zhiyuli on (3) "internal resources" means HydroShare resources.  For picking resources to add, the list should be any HydroShare resource the user has some form of access to.  I do not think the metadata class needs to change.  Right now a user can find the URL of a HydroShare resource and paste it in the box.  This is simply a convenience tool to save the user having to find, copy and paste the resource URL.  Regarding the display, once there is a URL, title and owners can be read, so can be entered in the display table.  I do not think there are any policy changes that are implied by this.  It is just making it easier for a user to add reference information (and hence not make mistakes), and presenting the reference information in a more usable form.

**Comment by pkdash** [comment]
@dtarb Check if this can be closed.

**Comment by dtarb** [comment]
Not all the functionality listed has been implemented, but to keep things manageable I have put unfinished items in #1494 and am closing.
